### ISSUE TITLE
But... Why?

### ISSUE THREAD

**Issue by quenbyako** [xakep666/monkey#3]
Monkey patching is a total evil in programming (except too specific cases, which are still not an argument)... For god sake, why did you publish that?

Also, looks like you didn't read original monkeypatching library [license](https://github.com/bouk/monkey/blob/master/LICENSE.md): 
> Don't use it.

Even author answered, why monkeypatching must not be exist in this world

----------------

I also must note you, that publishing projects in opensource is awesome thing, more opensourced projects — better life for developers. But it's better to keep code quality in opensource projects, cause if we won't look on code quality, opensource will become an evil too. Which is unacceptable/

**Comment by xhd2015** [comment]
Evil for what? Have you ever used it? Monkey patching is evil if you abuses it.But for testing, I don't see there is any problem with it.

**Comment by quenbyako** [comment]
@xhd2015 nope, using monkeypatching in tests is more evil than ever, cause if you do something like this in **even** in tests... Well, you got some problems with architecture for sure.

**Comment by xhd2015** [comment]
You don't even have experience about it.
